### PR TITLE
Add and remove

### PR DESCRIPTION
**Pull request by Sahar-AbdelSamad**
In this branch:
- We removed all items from the tasks array
- Created a new JavaScript file for the new functionality
- Implemented a function for adding a new task (add a new element to the array)
- Implemented a function for deleting a task (remove an element from the array)
- Implemented a function for editing task descriptions
All changes to the To Do List are saved in local storage

### CODE REVIEW

File: src/addAndRemove.js [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+      localStorage.setItem('list', JSON.stringify(tasks));
+      return;
+    }
+    tasks[i].index = i;
+    div[i].dataset.id = i;
+    const removeTask = document.querySelectorAll('.fa-ellipsis-v');
+    removeTask[i].id = i;
+    localStorage.setItem('list', JSON.stringify(tasks));
+  }
+};
```
Comment (mmsesay) [inline src/addAndRemove.js]: - [x] Good job writing this remove functionality. However, I noticed that the tasks can't be deleted. There is no indication like a trash icon or something else that will enable me to test out this functionality. I would appreciate you taking a look at it to figure out what might be the issue. Please verify that the functions work well before asking for a re-review.

File: src/addAndRemove.js [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+      div[ev.target.id].appendChild(inputCheckbox);
+      tasks[ev.target.id].description = editInput.value;
+      localStorage.setItem('list', JSON.stringify(tasks));
+      // Update Tasks
+      document.querySelectorAll('.fa-ellipsis-v').forEach((item) => item.addEventListener('click', editTaskFunction));
+    }
+  });
+  // Remove tasks
+  document.querySelector('.fa-trash-alt').addEventListener('click', removeTaskFunction);
+};
```
Comment (mmsesay) [inline src/addAndRemove.js]: - [x] Good job writing this editing functionality. However, I noticed that the tasks can't be edited, I tried double-clicking but it doesn't work.  I would appreciate you taking a look at it to figure out what might be the issue. Please verify that the functions work well before asking for a re-review.

File: src/addAndRemove.js [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+export const data = JSON.parse(localStorage.getItem('list'));
+let tasks = [];
+
+const id = 0;
+
+class Task {
+  constructor(description, complete, index) {
+    this.description = description;
+    this.complete = false;
+    this.index = id + 1;
```
Comment (mmsesay) [inline src/addAndRemove.js]: - [x] I can see you did set the index to start it index from 1. But it doesn't seem to take effect. When I add a task, your tasks index starts from 0. You should make the index starts from 1. And when you add a new task, we increment the index to the length of the current task + 1.

See example:
```
this.index =  tasks.length + 1;
```

**Review by mmsesay — CHANGES_REQUESTED** [review]
Hi @Sahar-AbdelSamad , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/addAndRemove.js (hunk 1 above)
Comment (Sahar-AbdelSamad) [reply src/addAndRemove.js]: Yes, there was a bug :+1:  I fixed it, thank you :smile:

File: src/addAndRemove.js (hunk 2 above)
Comment (Sahar-AbdelSamad) [reply src/addAndRemove.js]: Yes, there was a bug :+1:  I fixed it, thank you :smile:

File: src/addAndRemove.js (hunk 3 above)
Comment (Sahar-AbdelSamad) [reply src/addAndRemove.js]: Oh yes, I made it to start from 0 :see_no_evil: Thank you for pointing this out, I fixed it :smile:

**Comment by omar-labana** [comment]
Hi @Sahar-AbdelSamad, 

- [OPTIONAL] The live link in the README file is broken, please update it.

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**